index.bs

```diff
@@ -3253,7 +3253,7 @@ extended attribute’s [=takes a named argument list|named argument list=].
 </div>
 
 Two types are <dfn id="dfn-distinguishable" export>distinguishable</dfn> if
-at most one of the two [=includes a nullable type=]
+at most one of the two [=includes a nullable type=], is a [=dictionary type=], or a [=record type=],
```

Comment:
This is still somewhat ambiguous in the way that confused me in the first place. I don't see a simple way to fix that, but I've started a patch in https://github.com/heycam/webidl/compare/gh-pages...jyasskin:dictionary-distinguishability to try to make it clearer. I'll try to send a PR for that tomorrow, but feel free to steal it instead.